source/m2mresourceinstance.cpp

```diff
@@ -337,17 +337,13 @@ void M2MResourceInstance::get_value(uint8_t *&value, uint32_t &value_length)
     }
 }
 
-int M2MResourceInstance::get_value_int()
+int64_t M2MResourceInstance::get_value_int() const
```

Comment:
This still can't return 64-bit numbers, because it uses int and atoi.

Is _value zero-terminated now? It if isn't, this might be about 96% reliable.